packages/sdk/src/utils/helpers.ts

```diff
@@ -59,3 +59,7 @@ export function curry<T extends (...args: any[]) => any>(fn: T): Curry<T> {
   }
   return curried as Curry<T>
 }
+
+export function exhaustiveGuard(_value: never): never {
+  throw new Error(`Exhaustive guard reached with value ${_value}`)
```

Comment:
if a user receives this error will be so hard to debug

Reply:
Isn't this just for dev type checking?

runtime filters unhandled interfaces before this is ever hit:

<img width="635" alt="image" src="https://user-images.githubusercontent.com/3156233/235510426-71b5d93b-8cf5-4e1b-ae8c-12f68e89b518.png">

How do you suggest to improve?

Reply:
then if the exhaustive guards are never really hit what is the need of that implementation-level description in the changeset

Reply:
true, I can remove that, that's a dev detail